lib/NormalModule.js

```diff
@@ -212,6 +212,7 @@ class NormalModule extends Module {
 			rootContext: options.context,
 			webpack: true,
 			sourceMap: !!this.useSourceMap,
+			mode: options.mode,
```

Comment:
```suggestion
			mode: options.mode || "production",
```

Mode can be undefined. This is the case when you receive the NoModeWarning. Here we need to default to production.

Reply:
I didn't see this in the tests (where [no `mode` is supplied](https://github.com/webpack/webpack/pull/9140/commits/290ab5d0717ef89745459f5746cd8e598ee8f0da#diff-d5d0cb32d75d4327d1b9dcabb0ce42e7R1)).  I guess there are cases where the default is not applied?

Reply:
The test runner adds `mode: "production"` automatically, so that the tests are easier to write. This only happen when running webpack directly.

https://github.com/webpack/webpack/blob/62d1b15fe69bd08998d76d5210711e6217cd0aa8/test/ConfigTestCases.test.js#L72

Reply:
Got it.  Thanks for the pointer.